app.py

```diff
@@ -1,24 +1,116 @@
-from flask import url_for,render_template,redirect,Flask,flash
-from forms import LoginForm
+from flask import Flask, render_template, flash, redirect, url_for, session, request, logging, jsonify
+from passlib.hash import sha256_crypt
+from functools import wraps
+from forms import RegisterForm 
 from Make_Tables.mysqlconnect import mydb, mycursor, create_insert_statement   #Imported the mysqlconnect.py file from Make_tables folder
 
+
+
 app=Flask(__name__,static_url_path='/public')
 app.config['SECRET_KEY']='c828b6ff21f45063fd7860e5c1b1d233'
 
-@app.route('/')
-def home():
-    return render_template('index.html')
 
-@app.route("/login",methods=['GET','POST'])
-def Login():
-    form=LoginForm()
-    if form.validate_on_submit():
+
+# User Registration
+@app.route('/register', methods=['GET', 'POST'])
+def register():
+    form = RegisterForm(request.form)
+    if request.method == 'POST' and form.validate():
+        
+        userID = form.userID.data
         if form.email.data[1:6].isdecimal() and form.email.data[6:]=='@students.iitmandi.ac.in' :
-            flash('You have logged-in successfully!')
-            return redirect(url_for('home'))
+            email = form.email.data
+        name = form.name.data
+        password = sha256_crypt.encrypt(str(form.password.data))
```

Comment:
- [ ] password field in database is not big enough to hold the generated hash, edit database